Lib/test/test_uuid.py

```diff
@@ -675,6 +675,33 @@ def test_uuid_weakref(self):
         weak = weakref.ref(strong)
         self.assertIs(strong, weak())
 
+    @mock.patch.object(sys, "argv", ["", "-u", "uuid3", "-ns", "NAMESPACE_DNS"])
+    def test_cli_namespace_required_for_uuid3(self):
+        with self.assertRaises(SystemExit) as cm:
+            self.uuid.main()
+
+        # Check that exception code is the same as argparse.ArgumentParser.error
+        self.assertEqual(cm.exception.code, 2)
+
+    @mock.patch.object(sys, "argv", ["", "-u", "uuid3", "-n", "python.org"])
+    def test_cli_name_required_for_uuid3(self):
+        with self.assertRaises(SystemExit) as cm:
+            self.uuid.main()
+
+        # Check that exception code is the same as argparse.ArgumentParser.error
+        self.assertEqual(cm.exception.code, 2)
+
+    @mock.patch.object(sys, "argv", [""])
+    def test_cli_uuid4_outputted_with_no_args(self):
+        stdout = io.StringIO()
+        with contextlib.redirect_stdout(stdout):
+            self.uuid.main()
+
+        # Output uuid should be in the format of uuid4
+        self.assertRegex(
```

Comment:
I think instead we can pass the output to the `uuid.uuid4` function to be parsed. This is more clear and more correct.

Also, required params for other functions (namespace, name) are not tested. Let's add them!

Reply:
Hmm, I wasn't exactly sure of how to pass the output to `uuid.uuid4`, so instead I passed it to `uuid.UUID` and then checked string equality, and if the version was `4`. Hopefully that works? Also, added similar acceptance checks for the other functions requiring `namespace` and `name`!